rotkehlchen/chain/evm/node_inquirer.py

```diff
@@ -1169,20 +1183,62 @@ def get_erc20_contract_info(self, address: ChecksumEvmAddress) -> dict[str, Any]
             contract = EvmContract(address=address, abi=abi, deployed_block=0)
             decoded = self._process_contract_info(
                 output=output,
-                properties=properties,
+                properties=ERC20_PROPERTIES,
                 contract=contract,
                 token_kind=EvmTokenKind.ERC20,
             )
             log.debug(f'{address} was succesfuly decoded as ERC20 token')
 
-        for prop, value in zip_longest(properties, decoded):
+        for prop, value in zip_longest(ERC20_PROPERTIES, decoded):
             if isinstance(value, bytes):
                 value = value.rstrip(b'\x00').decode()  # noqa: PLW2901
             info[prop] = value
 
-        self.contract_info_erc20_cache[address] = info
         return info
 
+    def _query_token_contract(
+            self,
+            token_abi_str: Literal['ERC20_TOKEN', 'ERC721_TOKEN'],
+            properties: tuple[str, ...],
+            address: ChecksumEvmAddress,
+    ) -> list[tuple[bool, bytes]]:
+        abi = self.contracts.abi(token_abi_str)
```

Comment:
same comment about storing them in memory to avoid querying multiple times + json load